Add tt integrity check options (#4341)
tarantool/doc: doc/reference/tooling/tt_cli/start.rst

For detailed instructions on preparing and running Tarantool applications, see
:ref:`admin-instance-environment-overview` and :ref:`admin-start_stop_instance`.

See also: :ref:`tt-stop`, :ref:`tt-restart`, :ref:`tt-status`.

To start all instances of the application stored in the ``app`` directory inside
``instances_enabled`` in accordance with its ``instances.yml``:

..  code-block:: console

    $ tt start app

To start the ``router`` instance of the ``app`` application:

..  code-block:: console

    $ tt start app:router

When called without arguments, starts all enabled applications in the current environment:

> ..  code-block:: console
> 
>         $ tt start

Reviewer: The alignment is not the same as for previous commands.
```suggestion
..  code-block:: console

    $ tt start
```

    If you start such an application with ``tt start``, ``tt`` won't be able to check
    the application status or stop it using the corresponding commands.

.. _tt-start-integrity-check:

Integrity check
---------------

..  admonition:: Enterprise Edition
    :class: fact

    The integrity check functionality is supported by the `Enterprise Edition <https://www.tarantool.io/compare/>`_ only.

``tt start`` can perform initial and periodical integrity checks of the environment,
application, and centralized configuration.

To enable the integrity checks of environment and application files, you need to pack
the application using ``tt pack`` with the ``--with-integrity-check`` option.
This option generates and signs checksums of executables and configuration files in the current ``tt``
environment. Learn more in :ref:`tt-pack-integrity-check`.

> To enable the configuration integrity check, publish the configuration to a centralized

Reviewer: The configuration file in general will be packed with the application. So the local configuration is exist.
```suggestion
To add the configuration at the centralized storage integrity check, publish the configuration to a centralized
```

storage using ``tt cluster publish`` with the ``--with-integrity-check`` option.
This option generates and signs configuration checksums and saves them to the storage.
Learn more in :ref:`tt-cluster-publish-integrity`.

To perform the integrity checks when running the application, start it with the
``--integrity-check`` :ref:`global option <tt-global-options>`.
Its argument must be a public key matching the private key that was used for
generating checksums.

..  code-block:: console

    $ tt --integrity-check public.pem start myapp

After such a call, ``tt`` checks the environment, application, and configuration integrity
using the checksums and starts the application in case of the success. Then, integrity

> checks are performed periodically while the application is running. By default,

Reviewer: tarantool also checks hashes/signs of modules on the load time (require('module') is called).

    If you start such an application with ``tt start``, ``tt`` won't be able to check
    the application status or stop it using the corresponding commands.

.. _tt-start-integrity-check:

Integrity check
---------------

..  admonition:: Enterprise Edition
    :class: fact

    The integrity check functionality is supported by the `Enterprise Edition <https://www.tarantool.io/compare/>`_ only.

``tt start`` can perform initial and periodical integrity checks of the environment,
application, and centralized configuration.

To enable the integrity checks of environment and application files, you need to pack
the application using ``tt pack`` with the ``--with-integrity-check`` option.
This option generates and signs checksums of executables and configuration files in the current ``tt``
environment. Learn more in :ref:`tt-pack-integrity-check`.

To enable the integrity check of the configuration at the centralized storage,

> publish the configuration to a this storage using ``tt cluster publish`` with the ``--with-integrity-check`` option.

Reviewer 2: ```suggestion
publish the configuration to this storage using ``tt cluster publish`` with the ``--with-integrity-check`` option.
```
